packages/api/tests/endpoints/workflows.js

```diff
@@ -73,3 +107,70 @@ test('CUMULUS-912 GET with pathParameters and with an unauthorized user returns
     assertions.isUnauthorizedUserResponse(t, response);
   });
 });
+
+test.serial('GET with no path parameters and an authorized user returns a list of workflows', async (t) => {
+  const request = {
+    httpMethod: 'GET',
+    headers: authHeaders
+  };
+
+  return testEndpoint(workflowsEndpoint, request, (response) => {
+    t.is(response.statusCode, 200);
+    const results = JSON.parse(response.body);
+
+    t.deepEqual(results, workflowList);
+  });
+});
+
+test.serial('GET an existing workflow with an authorized user returns a specific workflow', async (t) => {
+  const request = {
+    httpMethod: 'GET',
+    pathParameters: {
+      name: 'HelloWorldWorkflow'
+    },
+    headers: authHeaders
+  };
+
+  return testEndpoint(workflowsEndpoint, request, (response) => {
+    t.is(response.statusCode, 200);
+    const result = JSON.parse(response.body);
+
+    t.deepEqual(result, workflowList[0]);
+    t.is(result.template, 's3://bucket/cumulus/workflows/HelloWorldWorkflow.json');
+  });
+});
+
+test.serial('GET with path parameters returns a 404 for a nonexistent workflow', async (t) => {
+  const request = {
+    httpMethod: 'GET',
+    pathParameters: {
+      name: 'NonexistentWorkflow'
+    },
+    headers: authHeaders
+  };
+
+  return testEndpoint(workflowsEndpoint, request, (response) => {
+    t.is(response.statusCode, 404);
+
+    const result = JSON.parse(response.body);
+    t.is(result.message, 'The specified key does not exist.');
+  });
+});
+
+test.serial('GET /good-workflow returns a 500 if the workflows list cannot be fetched from S3', async (t) => {
```

Comment:
Why are these tests serial? Because of how you're changing the `process.env.bucket` for this test and not wanting that to affect the other tests?

Reply:
I was trying to think of a way to break the dependency on `process.env.bucket` inside the endpoint handler code, but nothing is occurring to me. 

At the least, I think you only need to mark this test as `test.serial`, which I believe should make it run before any of the parallel tests [according to the AVA docs](https://github.com/avajs/ava/tree/v0.25.0#running-tests-serially).

Reply:
You could break the dependency by updating the workflow endpoint to look something like this:

```javascript
'use strict';

const _get = require('lodash.get');
const { S3 } = require('@cumulus/ingest/aws');
const handle = require('../lib/response').handle;

function list(event, bucket, cb) {
  const key = `${process.env.stackName}/workflows/list.json`;
  S3.get(bucket, key).then((file) => {
    const workflows = JSON.parse(file.Body.toString());
    return cb(null, workflows);
  }).catch((e) => cb(e));
}

function get(event, bucket, cb) {
  const name = _get(event.pathParameters, 'name');

  const key = `${process.env.stackName}/workflows/list.json`;
  S3.get(bucket, key)
    .then((file) => {
      const workflows = JSON.parse(file.Body.toString());

      const matchingWorkflow = workflows.find((workflow) => workflow.name === name);
      if (matchingWorkflow) return cb(null, matchingWorkflow);

      return cb({ message: `A record already exists for ${name}` });
    })
    .catch(cb);
}

function handleRequest(request, context, bucket) {
  handle(request, context, true, (cb) => {
    if (request.httpMethod === 'GET' && request.pathParameters) {
      get(request, bucket, cb);
    }
    else {
      list(request, bucket, cb);
    }
  });
}
module.exports.handleRequest = handleRequest;

function handleApiGatewayRequest(request, context) {
  handleRequest(request, context, process.env.bucket);
}
module.exports.handleApiGatewayRequest = handleApiGatewayRequest;
```

You would then configure API Gateway to use `handleApiGatewayRequest()` and run your tests to use `handleRequest()`.

Reply:
@yjpa7145 If you made the above changes, wouldn't you need to update `packages/api/index.js` like so?

```
exports.workflows = require('./endpoints/workflows').handleApiGatewayRequest;
```

Reply:
Also, I don't think we're saying this change is really necessary for this PR, but at minimum we should update so that only this test is running serially

Reply:
Oh that last one that changes the bucket is the only one that needs to be serial. I'll remove the label from the others.